mobile-app/app/screens/AppNavigator/screens/Loans/screens/PaybackLoanScreen.tsx

```diff
@@ -270,13 +273,13 @@ export function PaybackLoanScreen ({
         >
           <>
             <SetAmountButton
-              amount={new BigNumber(loanTokenAmount.amount ?? '0')}
+              amount={new BigNumber(loanTokenAmount.amount) > loanTokenBalanceInUSD ? loanTokenBalanceInUSD : new BigNumber('0')}
               onPress={onChangeFromAmount}
               type={AmountButtonTypes.half}
             />
 
             <SetAmountButton
-              amount={new BigNumber(loanTokenAmount.amount ?? '0')}
+              amount={new BigNumber(loanTokenAmount.amount) > loanTokenBalanceInUSD ? loanTokenBalanceInUSD : new BigNumber('0')}
```

Comment:
Validate against tokenBalance here, too. Also, should this be `new BigNumber('0')`? If the `loanTokenAmount.amount` is payable with your balance, we should input the whole `loanTokenAmount.amount` which is the total loan amount